src/common/collection.h

```diff
@@ -215,38 +187,16 @@ uint32_t dt_collection_get_query_flags(const dt_collection_t *collection);
 /** set filter flags for collection */
 void dt_collection_set_query_flags(const dt_collection_t *collection, uint32_t flags);
 
-/** get text filter for collection */
-char *dt_collection_get_text_filter(const dt_collection_t *collection);
-/** set text filter for collection */
-void dt_collection_set_text_filter(const dt_collection_t *collection, char *text_filter);
-
-/** get colors filter for collection */
-int dt_collection_get_colors_filter(const dt_collection_t *collection);
-/** set colors filter for collection */
-void dt_collection_set_colors_filter(const dt_collection_t *collection, int colors_filter);
-
 /** set the film_id of collection */
 void dt_collection_set_film_id(const dt_collection_t *collection, const int32_t film_id);
 /** set the tagid of collection */
 void dt_collection_set_tag_id(dt_collection_t *collection, const uint32_t tagid);
-/** set the star level for filter */
-void dt_collection_set_rating(const dt_collection_t *collection, uint32_t rating);
-/** get the star level for filter. The value returned starts on 0 **/
-uint32_t dt_collection_get_rating(const dt_collection_t *collection);
-/** set the comparator for rating */
-void dt_collection_set_rating_comparator(const dt_collection_t *collection,
-                                         const dt_collection_rating_comperator_t comparator);
-/** get the comparator for rating */
-dt_collection_rating_comperator_t dt_collection_get_rating_comparator(const dt_collection_t *collection);
-
-/** set the sort fields and flags used to show the collection **/
-void dt_collection_set_sort(const dt_collection_t *collection, dt_collection_sort_t sort, gint reverse);
-/** get the sort field used **/
-dt_collection_sort_t dt_collection_get_sort_field(const dt_collection_t *collection);
-/** get if the collection must be shown in descending order **/
-gboolean dt_collection_get_sort_descending(const dt_collection_t *collection);
+
```

Comment:
All of the above functions were not unused.  They were used by lua to interface with the collect module and get/set the state of the filters.

Reply:
yes, I know. but I've removed them because the system has completely changed, so this functions content are completely obsolete. I'm even not sure we can have the same interface.
About that, here are some thought : 
- is that ok for you if the function are located in filtering lib and not the collection.c ? (everything occurs there now)
- the number and type of filtering rules is not fixed anymore. How can we deal with that ? esp. what are we doing if lua request a value for a rule that is not present ?
- for sorting, that's the same, as we have multiple sort orders now...

I'm not sure what is the best in lua context... but for sure, if you need a function, just ask, I'll do it :)